### PR TITLE
Add an optional parameter similarity_score to VectorContextRetrieve…

### PR DESCRIPTION
**Pull request by ShashankVSG**
# Description

Added an optional parameter similarity_score to VectorContextRetriever for filtering the nodes retrieved

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-core/llama_index/core/indices/property_graph/sub_retrievers/vector.py [hunk 1]
```diff
@@ -137,8 +141,16 @@ def retrieve_from_graph(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
 
         assert len(triplets) == len(new_scores)
 
-        # sort by score
-        top_k = sorted(zip(triplets, new_scores), key=lambda x: x[1], reverse=True)
+        # filter by similarity score
+        if self._similarity_score:
+            filtered_data = [(triplet, score) for triplet, score in zip(triplets, scores) if score >= self._similarity_score]
```
Comment (nerdai) [inline llama-index-core/llama_index/core/indices/property_graph/sub_retrievers/vector.py:146]: should this be using `new_scores`?

Comment (nerdai) [inline llama-index-core/llama_index/core/indices/property_graph/sub_retrievers/vector.py:146]: should this be using new_scores?

File: llama-index-core/llama_index/core/indices/property_graph/sub_retrievers/vector.py [hunk 2]
```diff
@@ -190,7 +202,14 @@ async def aretrieve_from_graph(
...
+            # sort by score
+            top_k = sorted(filtered_data, key=lambda x: x[1], reverse=True)
+        else:
+            # sort by score
+            top_k = sorted(zip(triplets, new_scores), key=lambda x: x[1], reverse=True)
 
+        print(top_k)
```
Comment (nerdai) [inline llama-index-core/llama_index/core/indices/property_graph/sub_retrievers/vector.py:214]: can we remove this debug line?

File: llama-index-core/llama_index/core/indices/property_graph/sub_retrievers/vector.py [hunk 3]
```diff
@@ -35,6 +35,8 @@ class VectorContextRetriever(BasePGRetriever):
             The number of top similar kg nodes to retrieve. Defaults to 4.
         path_depth (int, optional):
             The depth of the path to retrieve for each node. Defaults to 1 (i.e. a triple).
+        similarity_score (int, optional):
```
Comment (nerdai) [inline llama-index-core/llama_index/core/indices/property_graph/sub_retrievers/vector.py:38]: should this be a float type instead?

**Comment by ShashankVSG** [comment]
@nerdai I had committed the test code by mistake. I've modified the code to resolve all the issues you've mentioned in the latest commit.

**Review by nerdai — APPROVED** [review]
lgtm!

@logan-markewich: quick check, do you forsee any issue with incorporating a `similarity_score` threshold in retrieval of `top_k` nodes?

**Review by logan-markewich — APPROVED** [review]

**Comment by logan-markewich** [comment]
@nerdai the only issue I can think of is if the vector store returns distance instead of a score.

**Comment by logan-markewich** [comment]
But, should be fine for now too :)

**Pull request merged**